### PR TITLE
Add GH80-3700

### PR DESCRIPTION
**Pull request by Maartenwut**
## Description

Adds the GH80-3700. The PCB has optional support for a PS/2 mouse, RGB strip or a speaker.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/maartenwut/gh80_3700/gh80_3700.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+ 
+#define LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/gh80_3700.h:29]:
```suggestion
#define LAYOUT_ortho_6x4( \
```

File: keyboards/maartenwut/gh80_3700/gh80_3700.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    k50,      k52, k53     \
+) \
+{ \
+    { k00, k01, k02, k03 },   \
+    { k10, k11, k12, k13 },   \
+    { k20, k21, k22, KC_NO }, \
+    { k30, k31, k32, k33 },   \
+    { k40, k41, k42, KC_NO }, \
+    { k50, KC_NO, k52, k53 }  \
+}
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/gh80_3700.h:61]:
```suggestion
}

#define LAYOUT_all LAYOUT_ortho_6x4
```

File: keyboards/maartenwut/gh80_3700/info.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{
+    "keyboard_name": "GH80-3700", 
+    "url": "https://github.com/Maartenwut/gh80-series/tree/master/GH80-3700", 
+    "maintainer": "Maartenwut", 
+    "width": 4, 
+    "height": 7, 
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/info.json:8]:
```suggestion
        "LAYOUT_ortho_6x4": {
```

File: keyboards/maartenwut/gh80_3700/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/default/keymap.c:19]:
```suggestion
[0] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/keymaps/ps2/keymap.c [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/ps2/keymap.c:19]:
```suggestion
[0] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/keymaps/rgb/keymap.c [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/rgb/keymap.c:19]:
```suggestion
[0] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c:19]:
```suggestion
[0] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  KC_ESC,   KC_TAB,   KC_EQL,   KC_BSPC,
+  KC_NLCK,  KC_PSLS,  KC_PAST,  KC_PMNS,
+  KC_P7,    KC_P8,    KC_P9,    KC_PPLS,
+  KC_P4,    KC_P5,    KC_P6,    KC_PPLS,
+  KC_P1,    KC_P2,    KC_P3,    KC_PENT,
+  KC_P0,    KC_P0,    KC_PDOT,  KC_PENT),
+  
+[1] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c:27]:
```suggestion
[1] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______),
+  
+[2] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c:35]:
```suggestion
[2] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______,
+  _______,  _______,  _______,  _______),
+  
+[3] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/via/keymap.c:43]:
```suggestion
[3] = LAYOUT_ortho_6x4(
```

File: keyboards/maartenwut/gh80_3700/rules.mk [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/rules.mk:21]:
```suggestion
```

File: keyboards/maartenwut/gh80_3700/rules.mk [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/rules.mk:23]:
```suggestion
```

File: keyboards/maartenwut/gh80_3700/rules.mk [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+ENCODER_ENABLE = yes        # Enable support for rotary encoders
+
+LAYOUTS = LAYOUT_numpad_6x4
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/rules.mk:26]:
```suggestion
LAYOUTS = numpad_6x4 ortho_6x4
```

File: keyboards/maartenwut/gh80_3700/keymaps/ps2/config.h [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (fauxpark) [inline keyboards/maartenwut/gh80_3700/keymaps/ps2/config.h:16]:
```suggestion

#pragma once

```

File: keyboards/maartenwut/gh80_3700/readme.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
+# GH80-3700
+
+![GH80-3700](https://i.imgur.com/xMCzb8v.jpg)
```
Comment (noroadsleft) [inline keyboards/maartenwut/gh80_3700/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![GH80-3700](https://i.imgur.com/xMCzb8vl.jpg)
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**